# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/ClientCanvasLayer.h
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f gfx/layers/client/ClientCanvasLayer.h
--- a/gfx/layers/client/ClientCanvasLayer.h	Wed Jun 10 21:57:19 2020 +0000
+++ b/gfx/layers/client/ClientCanvasLayer.h	Wed Jun 10 22:21:02 2020 +0000
@@ -30,7 +30,7 @@
     MOZ_COUNT_CTOR(ClientCanvasLayer);
   }
 
-  CanvasRenderer* CreateCanvasRendererInternal() override;
+  RefPtr<CanvasRenderer> CreateCanvasRendererInternal() override;
 
  protected:
   virtual ~ClientCanvasLayer();
@@ -59,7 +59,7 @@
   Layer* AsLayer() override { return this; }
   ShadowableLayer* AsShadowableLayer() override { return this; }
 
-  void Disconnect() override { mCanvasRenderer->Destroy(); }
+  void Disconnect() override { mCanvasRenderer->DisconnectClient(); }
 
   CompositableClient* GetCompositableClient() override {
     ClientCanvasRenderer* canvasRenderer =